File: web/regos-web/src/features/regulatory/registrations/types/RegistrationMarket.ts
/**
 * One country something is held in. Navigation, not analytics — the count says
 * whether a market is worth opening, nothing more.
 */
export interface RegistrationMarket {
  countryId: string;
  countryName: string;
  countryCode: string;
  registrationCount: number;
  /**
   * The regulatory groupings this market belongs to — EU, ICH, PIC/S. They
   * overlap, and **empty is a recorded answer**: India belongs to none of the
   * five RegOS records.
   */
  regions: string[];
}

File: src/Registration/RegOS.Registration.Application/Queries/ListRegistrationMarkets/RegistrationMarket.cs
namespace RegOS.Registration.Application.Queries.ListRegistrationMarkets;

/// <summary>
/// One country this tenant holds registrations in, and how many.
/// </summary>
/// <remarks>
/// <b>Navigation, not analytics.</b> The count exists so a market reads as
/// "Canada (12)" and a person can tell a busy market from a quiet one before
/// clicking — not as a metric. Breakdowns by status, trends and charts are
/// EPIC-011, and keeping this shape deliberately thin is what lets that arrive
/// without changing this contract.
/// </remarks>
/// <param name="Regions">
/// The groupings this market belongs to — what makes <em>"which of our markets
/// are in the EU?"</em> answerable from this one read rather than by joining
/// geography on the client.
/// </param>
public sealed record RegistrationMarket(
    Guid CountryId,
    string CountryName,
    string CountryCode,
    int RegistrationCount,
    IReadOnlyList<string> Regions);
